feat(Unity): Updated `options` and error capture explainer (#12904)
getsentry/sentry-docs: docs/platforms/unity/configuration/error-capture.mdx

---
title: Error Capture
sidebar_order: 60
description: "Learn more about how to configure your SDK to automatically send events to Sentry."
---

> The SDK relies on two integrations to capture errors: `UnityLogHandlerIntegration` and the `UnityApplicationLoggingIntegration`. While they sound similar, they serve different purposes.

Reviewer: ```suggestion
The Unity SDK relies on two integrations to capture errors: `UnityLogHandlerIntegration` and `UnityApplicationLoggingIntegration`. While they sound similar, they serve different purposes.
```

https://docs.unity3d.com/6000.0/Documentation/Manual/stack-trace.html

If configured, the SDK receives the stacktrace as part of the logging messages in a raw string format. The SDK is able to parse this format and display it in the Sentry UI. Unfortunately, to provide line numbers, the SDK needs to have an actual thrown exception object to fetch the required information for symbolication. Read more about the IL2CPP error symbolication works [here](/platforms/unity/configuration/il2cpp). If you're using Unity 6 or newer, you can enable the source code line numbers in the [player settings](https://docs.unity3d.com/6000.0/Documentation/Manual/il2cpp-managed-stack-traces.html). These line numbers are then part of the stringified stacktrace and will be parsed by the SDK.

## UnityApplicationLoggingIntegration

> The SDK uses the `UnityApplicationLoggingIntegration` to add its own log handler right before Unity's logging system and passes the logs back to Unity. This allows the SDK to capture errors from the `Application.logMessageReceived` method. All handled exceptions (i.e., those captured via `Log.LogException` calls) and unhandled exceptions are captured by the SDK. These messages contain the actual exception object, which allows the SDK to fetch the required information for symbolication.

Reviewer: ```suggestion
The SDK uses the `UnityApplicationLoggingIntegration` to add its own log handler right before Unity's logging system and passes the logs back to Unity. This allows the SDK to capture errors from the `Application.logMessageReceived` method. All handled exceptions (such as those captured via `Log.LogException` calls) and unhandled exceptions are captured by the SDK. These messages contain the actual exception object, which allows the SDK to fetch the required information for symbolication.
```

---
title: Error Capture
sidebar_order: 60
description: "Learn more about how to configure your SDK to automatically send events to Sentry."
---

The Unity SDK relies on two integrations to capture errors: `UnityLogHandlerIntegration` and `UnityApplicationLoggingIntegration`. While they sound similar, they serve different purposes.

## UnityLogHandlerIntegration

The SDK uses the `UnityLogHandlerIntegration` to hook into Unity's logging system. From there it receives all log messages that are logged via the `Debug.Log`, `Debug.LogWarning`, and `Debug.LogError` methods and those log messages are added as breadcrumbs to future events. By default, the SDK will also automatically capture messages logged via `Debug.LogError` as error events and send them to Sentry. This behaviour can be disabled by unchecking the automatic capture of `Debug.LogError` on the `Enrichment` tab, or programmatically, by setting the `CaptureLogErrorEvents` option to `false` in the [configure callback](/platforms/unity/configuration/options/programmatic-configuration).

> ### Stacktrace Support & Line Numbers

Reviewer 2: stack trace are two words
